### PR TITLE
Add PKI Init Script

### PR DESCRIPTION
**Pull request by goruha**
## what
* add script to easily init PKI if not already initialized
* use S6 to run both `openvpn` and `openvpn-api`

## why
* support the Cloud Posse chart for OpenVPN: https://github.com/cloudposse/charts/pull/20

## who
@cloudposse/engineering

### CODE REVIEW

File: Dockerfile [hunk 1]
```diff
@@ -62,4 +84,57 @@ RUN set -ex \
...
+          VERSION=$GITHUB_PAM_VERSION \
+          FILE=github-pam_linux_386 \
+          OUTPUT=github-pam-plugin \
+          make github:download-release  \
+      && chmod +x github-pam-plugin \
+      && mv github-pam-plugin /bin/ \
+      && apk add ca-certificates \
+      && apk del .build-deps; \
+    else \
+      echo '`GITHUB_TOKEN` required for fetching github-pam-plugin'; \
```
Comment (osterman) [inline Dockerfile]: I don't like that we need to expose a `GITHUB_TOKEN` to download *public* releases. This is a dependency we introduced originally when dealing with private repositories. I wish we didn't need to do it for public repos as it raises the barrier for others to contribute.

Comment (osterman) [inline Dockerfile]: Please deprecate `github-pam` since we're not supporting it and not releasing it.

File: Dockerfile [hunk 2]
```diff
@@ -62,4 +84,57 @@ RUN set -ex \
...
+      && REPO=cloudposse/openvpn-api \
+          VERSION=$OPENVPN_API_VERSION \
+          FILE=openvpn-api_linux_386 \
+          OUTPUT=openvpn-api \
+          make github:download-release \
+      && chmod +x openvpn-api \
+      && mv openvpn-api /bin/ \
+      && apk del .build-deps; \
+    else \
+      echo '`GITHUB_TOKEN` required for fetching openvpn-api'; \
```
Comment (osterman) [inline Dockerfile]: I don't like that we need to expose a `GITHUB_TOKEN` to download *public* releases. This is a dependency we introduced originally when dealing with private repositories. I wish we didn't need to do it for public repos as it raises the barrier for others to contribute.

File: rootfs/usr/local/bin/ovpn_genconfig [hunk 3]
```diff
@@ -0,0 +1,377 @@
+#!/bin/bash
```
Comment (osterman) [inline rootfs/usr/local/bin/ovpn_genconfig:1]: Why did we copy this file? What changed? Concerned about maintenance.

File: Dockerfile (hunk 2 above)
Comment (osterman) [reply Dockerfile]: Maybe we can modify `github:download-release` to `github:download-private-release` and `github:download-public-release`

File: Dockerfile [hunk 4]
```diff
@@ -2,42 +2,64 @@ FROM kylemanna/openvpn:latest
...
 
-ADD https://storage.googleapis.com/kubernetes-release/release/$K8S_VERSION/bin/linux/amd64/kubectl /usr/local/bin/kubectl
-RUN chmod +x /usr/local/bin/kubectl
-
 ARG GITHUB_TOKEN
-ARG REPO=cloudposse/github-pam
-ARG FILE=github-pam_linux_386
-ARG VERSION=0.10
 ARG PAM_SCRIPT_VERSION=1.1.8-1
+ARG S6_OVERLAY_VER=1.17.2.0
```
Comment (osterman) [inline Dockerfile]: Inconsistently named. Should be `S6_OVERLAY_VERSION`

**Review by osterman — CHANGES_REQUESTED** [review]
See inline comments

File: Dockerfile [hunk 5]
```diff
@@ -2,42 +2,64 @@ FROM kylemanna/openvpn:latest
...
+ENV DUO_FAILMODE=secure
+ENV DUO_AUTOPUSH=yes
+ENV DUO_PROMPTS=1
 
-RUN if [ ! -z $GITHUB_TOKEN ]; then \
-      set -ex \
-      && echo "http://dl-cdn.alpinelinux.org/alpine/edge/testing" >> /etc/apk/repositories \
-      && apk update \
-      && apk add --no-cache --virtual .build-deps \
+ENV GITHUB_PAM_TEAM=
```
Comment (osterman) [inline Dockerfile]: Deprecate

File: rootfs/usr/local/bin/ovpn_genconfig (hunk 3 above)
Comment (goruha) [reply rootfs/usr/local/bin/ovpn_genconfig:1]: Original file does not support OVPN_RENEG_SEC option

Comment (goruha) [reply rootfs/usr/local/bin/ovpn_genconfig:1]: can we just add openvpn docker image releases (push to hub.docker) to software updates channel?

Comment (osterman) [reply rootfs/usr/local/bin/ovpn_genconfig:1]: I don't follow the question....

File: Dockerfile (hunk 2 above)
Comment (osterman) [reply Dockerfile]: I am also okay with just straight-up curl'ing the public release

File: Dockerfile (hunk 1 above)
Comment (goruha) [reply Dockerfile]: Done

File: rootfs/usr/local/bin/ovpn_genconfig (hunk 3 above)
Comment (goruha) [reply rootfs/usr/local/bin/ovpn_genconfig:1]: to get notifications when openvpn image new version release

File: Dockerfile (hunk 4 above)
Comment (goruha) [reply Dockerfile]: Done

File: Dockerfile (hunk 5 above)
Comment (goruha) [reply Dockerfile]: Done

File: Dockerfile (hunk 2 above)
Comment (goruha) [reply Dockerfile]: Done

File: Dockerfile [hunk 6]
```diff
@@ -2,64 +2,76 @@ FROM kylemanna/openvpn:latest
 
 ARG K8S_VERSION=v1.5.1
 
+ARG GITHUB_TOKEN
```
Comment (osterman) [inline Dockerfile]: Deprecate `GITHUB_TOKEN`?

**Review by osterman — APPROVED** [review]
Drop `GITHUB_TOKEN`, then merge

File: Dockerfile (hunk 6 above)
Comment (goruha) [reply Dockerfile]: Done

**Pull request merged**